### PR TITLE
Fixed window close button on some FPGA windows

### LINKED ISSUE #1852: FPGA Board Editor Properties windows does not close with 'x'
Using:
Logisim Evolution 3.8
Windows 11 Home build 22621

On the FPGA Board Editor when clicking on an item, such as a pin, the property menu comes up in another window. The issue is that in this window the 'X' in the top right does not do anything. Expected functionally was to close the sub window.
This seems to happen regardless of the item type. Tested with: Pin, portIO, LED Array, 7 seg, and DIP switch.

Button in question is highlighted in yellow:
image

### PR DESCRIPTION
**Pull request by aplq**
This resolves #1852

### CODE REVIEW

File: src/main/java/com/cburch/logisim/fpga/gui/FpgaIoInformationSettingsDialog.java [hunk 1]
```diff
@@ -712,11 +714,17 @@ else if (cnt < nrOfPins.get(INPUT_ID) + nrOfPins.get(OUTPUT_ID))
...
+      @Override
+      public void windowClosing(WindowEvent e) {
+        CancelButton.doClick();
+      }
+    });
     selWindow.add(new JScrollPane(contents));
     selWindow.pack();
     selWindow.setLocationRelativeTo(IOcomps.getParentFrame());
     selWindow.setModal(true);
-    selWindow.setDefaultCloseOperation(JDialog.DO_NOTHING_ON_CLOSE);
```
Comment (maehne) [inline src/main/java/com/cburch/logisim/fpga/gui/FpgaIoInformationSettingsDialog.java:719]: Why you delete it here and you don't do the same in line 1102 below (you need to expand the view)?

**Review by maehne — COMMENTED** [review]
Looks mostly good to me. I primarily wonder why line 719 gets deleted, but not line 1102. I may be overlooking something though.

@BFH-ktt1: Could you please have a closer look?

File: src/main/java/com/cburch/logisim/fpga/gui/FpgaIoInformationSettingsDialog.java (hunk 1 above)
Comment (aplq) [reply src/main/java/com/cburch/logisim/fpga/gui/FpgaIoInformationSettingsDialog.java:719]: Looking into it I probably should not have removed it. Both of the 'x' buttons work, so having it in there is not breaking the new functionality. I will add it back.
I have not worked with GUIs before.

**Review by maehne — COMMENTED** [review]
@davidhutchens, @MarcinOrlowski, or @BFH-ktt1: Could someone of you have a look on this PR and provide feedback if you see a more elegant way?

**Comment by davidhutchens** [comment]
@maehne: It might be that setDefaultCloseOperation to one of DISPOSE_ON_CLOSE or HIDE_ON_CLOSE would provide what is needed without the WindowAdapter.

But the original code probably sets it to DO_NOTHING_ON_CLOSE for a reason, such as some cleanup is needed that the provided buttons perform. If Cancel does some needed actions, then it is appropriate to write the code with WindowAdapter as @aplq has done here. I also concur with @aplq in leaving the DISPOSE_ON_CLOSE settings in the code when using a WindowAdapter.

@aplq: If you are satisfied that you have addressed @maehne's concern, you can use the "Resolve Conversation" button to close the concern.

**Review by BFH-ktt1 — APPROVED** [review]

**Pull request merged**